cg/services/order_validation_service/workflows/tomte/validation/inter_field_validators.py

```diff
@@ -0,0 +1,52 @@
+from cg.models.orders.sample_base import ContainerEnum
+from cg.services.order_validation_service.models.errors import OccupiedWellError, ValidationError
+from cg.services.order_validation_service.workflows.tomte.models.order import TomteOrder
+from cg.services.order_validation_service.workflows.tomte.models.sample import TomteSample
+from cg.services.order_validation_service.workflows.tomte.models.case import TomteCase
+
+
+def validate_wells_contain_at_most_one_sample(order: TomteOrder) -> list[OccupiedWellError]:
+    plate_samples_with_cases: list[tuple[TomteSample, str]] = _get_plate_samples(order)
+    colliding_samples: list[tuple[TomteSample, str]] = _get_colliding_samples(
+        plate_samples_with_cases
+    )
+    return _get_errors(colliding_samples)
+
+
+def _get_errors(colliding_samples: list[tuple[TomteSample, str]]) -> list[OccupiedWellError]:
+    errors: list[OccupiedWellError] = []
+    for sample, case_id in colliding_samples:
+        error = OccupiedWellError(sample_id=sample.internal_id, case_id=case_id)
+        errors.append(error)
+    return errors
+
+
+def _get_plate_samples(order: TomteOrder) -> list[tuple[TomteSample, str]]:
+    return [
+        (sample, case.internal_id)
+        for case in order.cases
+        for sample in case.samples
+        if sample.container == ContainerEnum.plate
+    ]
+
+
+def _get_colliding_samples(
+    samples_with_cases: list[tuple[TomteSample, str]]
+) -> list[tuple[TomteSample, str]]:
+    colliding_samples = []
+    sample_well_map = _get_sample_well_map(samples_with_cases)
+    for _, well_samples in sample_well_map.items():
+        if len(well_samples) > 1:
+            colliding_samples.extend(well_samples[1:])
```

Comment:
It is unclear why you only choose the last samples. Shouldn't you choose all samples that are in one well? I guess the order is arbitrary

Reply:
The reason is that a well can contain one sample but not more. So one of the samples is allowed to be in the well, but all others need to be moved to other wells.

Reply:
Extracted it to a variable with a descriptive name https://github.com/Clinical-Genomics/cg/pull/3437/commits/82c8160c4929ee649237c44d905e37e58f2c2318